core/server/src/main/java/alluxio/worker/netty/DataServerReadHandler.java

```diff
@@ -72,6 +73,9 @@
   @GuardedBy("mLock")
   private long mPosToWrite = -1;
 
+  /** Makes sure we send EOF only once. */
+  private final AtomicBoolean mEOFSent = new AtomicBoolean(false);
```

Comment:
good point. I can move the mEOFSent reset to channelRead which should solve the problem though i don't feel that is clean.